src/cargo/ops/cargo_rustc/fingerprint.rs

```diff
@@ -148,11 +148,10 @@ impl Fingerprint {
     fn update_local(&self) -> CargoResult<()> {
         match self.local {
             LocalFingerprint::MtimeBased(ref slot, ref path) => {
-                let meta = fs::metadata(path).chain_error(|| {
-                    internal(format!("failed to stat `{}`", path.display()))
-                })?;
-                let mtime = FileTime::from_last_modification_time(&meta);
-                *slot.0.lock().unwrap() = Some(mtime);
+                if let Ok(meta) = fs::metadata(path) {
+                    let mtime = FileTime::from_last_modification_time(&meta);
+                    *slot.0.lock().unwrap() = Some(mtime);
+                }
```

Comment:
Hm I'd prefer to not change this in exactly this manner. This seems like it's just waiting to accidentally ignore errors and then not cause rebuilds when they should happen otherwise. Could such an error like this be caught elsewhere and handled?

For example I could imagine something higher up catching this error and leaving behind a tombstone for "this must be recompiled", or something like that.